rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/condition/ConditionGenerator.xtend

```diff
@@ -109,7 +109,18 @@ class ConditionGenerator {
 					private «ComparisonResult» executeDataRule(«rosettaClass.name» «defaultClassExecuteScope.createIdentifier(implicitVarRepr, rosettaClass.name.toFirstLower)») {
 						try {
 							«ComparisonResult» «defaultClassExecuteResultId» = «expressionHandler.toComparisonResult(rule.expression, defaultClassExecuteScope)»;
-							return «defaultClassExecuteResultId».get() == null ? ComparisonResult.success() : «defaultClassExecuteResultId»;
+							if («defaultClassExecuteResultId».get()) {
```

Comment:
Shouldn't this be `if («defaultClassExecuteResultId».get() == null) {`

Reply:
i am replicating «defaultClassExecuteResultId».get() == null ? ComparisonResult.success() : «defaultClassExecuteResultId»; where we were returning ComparisonResult.success()